File: ARCADE/Intro/Level 8_4: arrayMaxConsecutiveSum.cs
    using System;
     
    public class Test
    {
    	public static int arrayMaxConsecutiveSum(int[] inputArray, int k){
            int result = 0, sum = 0;
            for (int i = 0; i < k - 1; i++){
                sum += inputArray[i];
            }
            for (int i = k - 1; i < inputArray.Length; i++){
                sum += inputArray[i];
                if (sum > result){
                    result = sum;
                }
                sum -= inputArray[i - k + 1];
            }
            return result;
        }
    	public static void Main()
    	{
    		int [] inputArray = new int []{2, 3, 5, 1, 6};
    		int k = 2;
    		Console.WriteLine(arrayMaxConsecutiveSum(inputArray, k));
    	}
    }

File: ARCADE/Intro/Level 8_4: arrayMaxConsecutiveSum.pas
program arrayMaxConsecutiveSum;

Type
	ArrayInteger = Array of integer;

function arrayMaxConsecutiveSum(inputArray: ArrayInteger; k: integer): integer;
var i, j, sum, max : integer;
begin
    max := 0; sum := 0;
    for i := 0 to k - 2 do
        sum := sum + inputArray[i];
        
    for i := k - 1 to length(inputArray) - 1 do
    begin
        sum := sum + inputArray[i];
        if(sum > max) then max := sum;
        sum := sum - inputArray[i-k+1];
    end;
    
    arrayMaxConsecutiveSum := max;
end;

var inputArray : ArrayInteger;
	k : integer;
begin
	k := 2;
	inputArray := ArrayInteger.Create(2, 3, 5, 1, 6);
	writeln(arrayMaxConsecutiveSum(inputArray, k));
end.
